src/table.rs

```diff
@@ -77,15 +86,24 @@ pub(crate) trait Slot: Any + Send + Sync {
     /// The current revision MUST be the current revision of the database containing this slot.
     unsafe fn memos(&self, current_revision: Revision) -> &MemoTable;
 
-    /// Mutably access the [`MemoTable`] for this slot.
-    fn memos_mut(&mut self) -> &mut MemoTable;
+    /// Access the [`MemoTable`] for this slot.
+    ///
+    /// # Safety
+    ///
+    /// You must have mutable access to the slot.
+    unsafe fn memos_no_revision(&self) -> &MemoTable;
```

Comment:
Then why does this not just take `&mut self`

Reply:
Problems with the borrow checker when calling `Page::get()` borrows the entire page then we cannot access `Page::memo_types`.